src/app/services/notification.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { AngularFirestore, AngularFirestoreCollection } from '@angular/fire/firestore';
+import Notification from '../models/notification';
+import { PostService } from './post.service';
+import { UserService } from './user.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NotificationService {
+
+  private notificationPath : string = 'notifications';
+  notificationCollection: AngularFirestoreCollection<Notification>;
+
+  constructor(
+    private db: AngularFirestore,
+  ) {
+    this.notificationCollection = db.collection<Notification>(this.notificationPath);
+  }
+
+  observable(userID: string, handleAdd?: Function, handleModification?:Function, handleDelete?:Function) {
+    return this.notificationCollection.ref.where("userID", "==", userID).onSnapshot((snapshot) => {
+        snapshot.docChanges().forEach((change) => {
+          const doc = change.doc.data();
+            if (handleAdd !== undefined && change.type === "added") {
+                handleAdd(doc)
+            }
+           else if (handleModification !== undefined && change.type === "modified") {
```

Comment:
Will we ever need to listen for updates to a notification? I feel notifications should just be a one-time message and shouldn't be editable.

I'm guessing you added this for the case of a notification being "marked as read"? We wouldn't need to listen for that right, since only the user who marked it as read would be able to mark it as read.  So, instead we could just update the UI directly after they click "Mark all as read" instead of listening for it.

Reply:
Good point. Should we also have the same behavior for deleting notifications? I assume only the user themselves would be able to delete notifications.

Reply:
Yeah, same behaviour for deletes makes sense too.